examples/trainer/simple_auto_trainer/notebook.ipynb

```diff
@@ -0,0 +1,303 @@
+    " \n",
+    "We will use a preprocessed version of the IMDb dataset. In https://github.com/knodle/knodle/examples, you can find a tutorial showing you how the data was preprossed and transformed into the Knodle format. Instead of using the download in the cells below, you can also use this tutorial to create the data yourself."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "frank-economy",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext autoreload\n",
+    "%autoreload 2\n",
+    "\n",
+    "import os\n",
+    "\n",
+    "imdb_data_dir = os.path.join(os.getcwd(), \"data\", \"imdb\")\n",
+    "processed_data_dir = os.path.join(imdb_data_dir, \"processed\")\n",
+    "os.makedirs(processed_data_dir, exist_ok=True)\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "silver-venice",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from minio import Minio\n",
+    "from tqdm.auto import tqdm\n",
+    "\n",
+    "client = Minio(\"knodle.dm.univie.ac.at\", secure=False)\n",
+    "files = [\n",
+    "    \"df_train.csv\", \"df_dev.csv\", \"df_test.csv\",\n",
+    "    \"train_rule_matches_z.lib\", \"dev_rule_matches_z.lib\", \"test_rule_matches_z.lib\",\n",
+    "    \"mapping_rules_labels_t.lib\"\n",
+    "]\n",
+    "\n",
+    "for file in tqdm(files):\n",
+    "    client.fget_object(\n",
+    "        bucket_name=\"knodle\",\n",
+    "        object_name=os.path.join(\"datasets/imdb/processed\", file),\n",
+    "        file_path=os.path.join(processed_data_dir, file),\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "natural-slope",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import joblib\n",
+    "import pandas as pd\n",
+    "\n",
+    "df_train = pd.read_csv(os.path.join(processed_data_dir, \"df_train.csv\"))\n",
+    "df_dev = pd.read_csv(os.path.join(processed_data_dir, \"df_dev.csv\"))\n",
+    "df_test = pd.read_csv(os.path.join(processed_data_dir, \"df_test.csv\"))\n",
+    "\n",
+    "mapping_rules_labels_t = joblib.load(os.path.join(processed_data_dir, \"mapping_rules_labels_t.lib\"))\n",
+    "\n",
+    "train_rule_matches_z = joblib.load(os.path.join(processed_data_dir, \"train_rule_matches_z.lib\"))\n",
+    "dev_rule_matches_z = joblib.load(os.path.join(processed_data_dir, \"dev_rule_matches_z.lib\"))\n",
+    "test_rule_matches_z = joblib.load(os.path.join(processed_data_dir, \"test_rule_matches_z.lib\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "operational-presence",
+   "metadata": {},
+   "source": [
+    "### Data description\n",
+    "\n",
+    "We have three splits: train, develop and test split. For each split, there is DataFrame, holding text, and a Z matrix, relating instances, or rows in the DataFrame, to rules. Again, for more information we refer to the creation of the Dataset https://github.com/knodle/knodle/example.\n",
```

Comment:
"We have three splits: train, develop and test split. For each split, there is:
- a DataFrame, holding text (for training set) or text&labels (for development and test set)
- a Z matrix, relating instances, or rows in the DataFrame, to rules. 

Again, for more information we refer to the creation of the Dataset https://github.com/knodle/knodle/example.\n",